app/models/service/resource_linking.rb

```diff
@@ -0,0 +1,37 @@
+module Service::ResourceLinking
+  extend ActiveSupport::Concern
+
+  def add_provider_vms_queue(provider, array_of_uid_ems)
```

Comment:
So this implementation intents to add resources by type? Next time if we want to link networks, we need to introduce new method `add_provider_networks_queue` and copy the same logic?